### PR TITLE
Smart storage

### PR DESCRIPTION
**Pull request by TC-DarkSlayer**
- Fixed Jammer's SmartStorage
- Organized Builder Menu
- Added Iron Ladder

### CODE REVIEW

File: TFlippy_TerritoryControl_Core_Dev/Entities/Structures/SmartStorageHelpers.as [hunk 1]
```diff
@@ -0,0 +1,53 @@
+//returns amount of this resource in inventory
+u32 smartStorageCheck(CBlob@ this, string iname)
+{
+	int64 ret;
+	dictionary@ inventory;
+	if(!this.get("smart_inventory", @inventory) || !inventory.get(iname,ret))
+		return 0;
+	return ret;
```
Comment (Vam-Jam) [inline TFlippy_TerritoryControl_Core_Dev/Entities/Structures/SmartStorageHelpers.as:8]: you are returning an int64 as a u32

File: TFlippy_TerritoryControl_Core_Dev/Entities/Structures/SmartStorageHelpers.as [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	}
+	this.set_u16("smart_storage_quantity",cur_quantity);
+	server_Sync(this, iname, am, mq, cur_quantity);
+	return amount;
+}
+
+// KAG's CBlob.Sync() is nonfunctional shit <- ???
+void server_Sync(CBlob@ this, string iname, u32 new_amount, u16 max_quantity, u16 new_quantity)
+{
+	if (getNet().isServer())
```
Comment (Vam-Jam) [inline TFlippy_TerritoryControl_Core_Dev/Entities/Structures/SmartStorageHelpers.as:44]: isServer()

File: TFlippy_TerritoryControl_Core_Dev/Entities/Structures/SmartStorageHelpers.as [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	}
+	else
+	{
+		am -= amount;
+		amount = 0;
+		inventory.set(iname, am);
+		cur_quantity-=(prevstacks-((am-1)/mq+1));
+	}
+	this.set_u16("smart_storage_quantity",cur_quantity);
+	server_Sync(this, iname, am, mq, cur_quantity);
```
Comment (Vam-Jam) [inline TFlippy_TerritoryControl_Core_Dev/Entities/Structures/SmartStorageHelpers.as:37]: am should be u32, not int64, mq should be u16 not int64

**Review by Vam-Jam — CHANGES_REQUESTED** [review]
Far too many inconsistencies with in64 being returned as u32/u16 and sending int64 as a u16, please fix them.

**Review by laikadoggirl — CHANGES_REQUESTED** [review]
Also nerfed a bunch of buildings in here as well, such as concrete blocks requiring twice as much, another drill rig nerf, and another hopper nerf

Also changed capturables in a smart storage commit?

**Pull request closed without merging**